yt/utilities/lib/pixelization_routines.pyx

```diff
@@ -1284,6 +1284,191 @@ def pixelize_sph_kernel_projection(
+            h_j2 = fmax(hsml[j]*hsml[j], dx*dy)
+            ih_j2 = 1.0/h_j2
+
+            prefactor_j = pmass[j] / pdens[j] / hsml[j]**2 * quantity_to_smooth[j]
+            if weight_field is not None:
+                prefactor_j *= _weight_field[j]
+
+            for ii in range(2):
+                if xiter[ii] == 999: continue
+                px = posx[j] + xiterv[ii]
+                if (px + hsml[j] < x_min) or (px - hsml[j] > x_max): continue
+
+
+                for jj in range(2):
+                    if yiter[jj] == 999: continue
+                    py = posy[j] + yiterv[jj]
+                    if (py + hsml[j] < y_min) or (py - hsml[j] > y_max): continue
+
+                    for kk in range(2):
+                        # discard if z is outside bounds
+                        if ziter[kk] == 999: continue
+                        pz = posz[j] + ziterv[kk]
+                        if (pz + hsml[j] < z_min) or (pz  - hsml[j] > z_max): continue
+
+                        # here we find the pixels which this particle contributes to
+                        x0 = <np.int64_t> ((px - hsml[j] - x_min)*idx)
+                        x1 = <np.int64_t> ((px + hsml[j] - x_min)*idx)
+                        x0 = iclip(x0-1, 0, xsize)
+                        x1 = iclip(x1+1, 0, xsize)
+
+                        y0 = <np.int64_t> ((py - hsml[j] - y_min)*idy)
+                        y1 = <np.int64_t> ((py + hsml[j] - y_min)*idy)
+                        y0 = iclip(y0-1, 0, ysize)
+                        y1 = iclip(y1+1, 0, ysize)
+
+                        # found pixels we deposit on, loop through those pixels
+                        for xi in range(x0, x1):
+                            # we use the centre of the pixel to calculate contribution
+                            x = (xi + 0.5) * dx + x_min
+
+                            posx_diff = px - x
+                            posx_diff = posx_diff * posx_diff
+
+                            if posx_diff > h_j2: continue
+
+                            for yi in range(y0, y1):
+                                y = (yi + 0.5) * dy + y_min
+
+                                posy_diff = py - y
+                                posy_diff = posy_diff * posy_diff
+                                if posy_diff > h_j2: continue
+
+                                q_ij2 = (posx_diff + posy_diff) * ih_j2
+                                if q_ij2 >= 1:
+                                    continue
+
+                                # see equation 32 of the SPLASH paper
+                                # now we just use the kernel projection
+                                local_buff[xi + yi*xsize] +=  prefactor_j * itab.interpolate(q_ij2)
+                                mask[xi, yi] = 1
+
+        with gil:
+            for xxi in range(xsize):
+                for yyi in range(ysize):
+                    buff[xxi, yyi] += local_buff[xxi + yyi*xsize]
+        free(local_buff)
+        free(xiterv)
```

Comment:
does this also need to happen inside the parallel context?

Reply:
hard to tell with the github interface, but these lines are all within the 

```
with nogil, parallel():
```

block way up top. 

<img width="612" alt="Screenshot 2023-11-13 at 11 09 45 AM" src="https://github.com/yt-project/yt/assets/22038879/93b809a3-7983-44c0-b0fc-ec12d44d77eb">